### DISCUSSION TITLE
Discard a geometry inside 4 points

### DISCUSSION THREAD

**Discussion by Daoudou** [openscenegraph/OpenSceneGraph#1230] [Q&A]
Hello everyone

I would need help.

I have a surface created with geometry and I would like to cut out part of the surface to place an object on it without the surface being inside the object (I have an open boat and the blue surface crosses the boat and is therefore visible inside). (see picture)

![Capture4](https://github.com/openscenegraph/OpenSceneGraph/assets/98310378/c63eed4e-a654-42cb-99a2-2259f843b787)

For this I have 4 corner points and I would like to delete the geometry and primitives located inside these 4 corners. (the white square on the image)

For this how could I proceed?

A solution would be to obtain the position of these 4 points in a shader and to delete the fragments inside with the "discrad" method in glsl if I'm not mistaken? (However I don't have enough knowledge with shaders ^^')

Or do you have any other suggestions?

i found this source : 

[osg-users] [osgOcean] Culling and intersects with ocean and model...

Thanks in advance
Cordially

**Comment by robertosfield** [comment]
I can't figure out specifically what you need from your description so it's not possible to give specific advice.

Is it just rendering that you need to alter, or do you need to modify the mesh and vertex data or geometry?

How is the original geometry created?  How is the hole controlled?

**Reply by Daoudou** [reply]
hello

I need to modify only the rendering, for the original geometry I only have a cube and for the hole I pass specific coordinates manually for the 4 points.

**Comment by robertosfield** [comment]
Stencil buffer might be useful for doing cutouts.  You'd render the cutout geometry to the stencil buffer then in main pass check the stencil buffer to decide which fragments to render or not.  You can do something similar with use render to texture.

**Reply by Daoudou** [reply]
Okay and if i want to do it with only to modify the mesh and vertex data ? It will work the same way ? With Stencil buffer ?

**Comment by LaurensVoerman** [comment]
You might be able to render the water surface first, and not write into the
depth buffer, so anything rendered later will overwrite the pixels. Here is
an example in that vein:
(it does write the depth buffer , but always "1" (far clip plane)
Regards, Laurens.

    StateSet {
      DataVariance STATIC
      rendering_hint DEFAULT_BIN
      renderBinMode USE
      binNumber -1
      binName RenderBin
      GL_CULL_FACE OVERRIDE|OFF
      GL_LIGHTING OVERRIDE|OFF
      GL_DEPTH_TEST ON
      Depth {
        function ALWAYS
        writeMask TRUE
        range 1 1
      }
    }
